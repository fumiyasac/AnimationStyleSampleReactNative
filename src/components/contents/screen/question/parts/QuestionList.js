/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 * @flow
 */

import React, { Component } from 'react';

import {
  StyleSheet
} from 'react-native';

import {
  Content
} from 'native-base';

// よくある質問コンテンツ用の共通コンポーネント
import QuestionListIntro from './QuestionListIntro';
import QuestionListItem from './QuestionListItem';

export default class QuestionList extends React.Component {

  // MARK: - Functions

  // 取得したよくある質問コンテンツ用データを表示する処理
  _displayQuestionList = () => {
    const { questionItems } = this.props;

    // よくある質問コンテンツ用データのリストを生成して表示する
    return questionItems.items.map( (item, index) => {
      return (
        <QuestionListItem key={index} item={item} />
      );
    });
  };

  // MARK: - Rendering Components

  render() {
    return (
      <Content style={styles.container}>
        <QuestionListIntro />
        {this._displayQuestionList()}
      </Content>
    );
  };
}

// MARK: - Component Styles

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#ffffff'
  }
});
