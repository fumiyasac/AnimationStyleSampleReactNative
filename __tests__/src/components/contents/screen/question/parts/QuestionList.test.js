import 'react-native';
import React from 'react';
import QuestionList from '../../../../../../../src/components/contents/screen/question/parts/QuestionList';

import renderer from 'react-test-renderer';

it('QuestionListコンポーネントが正しく表示されること', () => {

	let stubObject = {
		"items": [
			{ 
				title: "Q. このサンプルに改変や改良をしても大丈夫ですか？", 
				content: "A. ご自分でデザインやアカウントが必要な部分を改変した上でご活用頂けましたら、特に問題はないかと思います。もし自分の環境で動かなかった等があればGithubのissue等へ記載をしていただければ幸いです。しかしながら回答によってはご対応が出来かねます場合があることも了承頂ければと思います。"
			}
		]
	};

  const tree = renderer.create(
		<QuestionList questionItems={stubObject} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});