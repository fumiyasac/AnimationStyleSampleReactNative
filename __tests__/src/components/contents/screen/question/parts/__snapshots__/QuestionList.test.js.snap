// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuestionListコンポーネントが正しく表示されること 1`] = `
<RCTScrollView
  automaticallyAdjustContentInsets={false}
  contentContainerStyle={
    Array [
      Object {
        "padding": undefined,
      },
      undefined,
    ]
  }
  contentInset={
    Object {
      "bottom": 0,
    }
  }
  enableAutomaticScroll={true}
  enableOnAndroid={false}
  enableResetScrollToCoords={true}
  extraHeight={75}
  extraScrollHeight={0}
  getScrollResponder={[Function]}
  handleOnScroll={[Function]}
  keyboardDismissMode="interactive"
  keyboardOpeningTime={250}
  keyboardShouldPersistTaps="handled"
  keyboardSpace={0}
  onScroll={[Function]}
  resetKeyboardSpace={[Function]}
  resetScrollToCoords={
    Object {
      "x": 0,
      "y": 0,
    }
  }
  scrollEventThrottle={1}
  scrollForExtraHeightOnAndroid={[Function]}
  scrollIntoView={[Function]}
  scrollToEnd={[Function]}
  scrollToFocusedInput={[Function]}
  scrollToPosition={[Function]}
  showsVerticalScrollIndicator={true}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
        "flex": 1,
      },
      Object {
        "backgroundColor": "#ffffff",
        "flex": 1,
      },
    ]
  }
  update={[Function]}
  viewIsInsideTabBar={false}
  virtual={undefined}
>
  <View>
    <View>
      <Text
        style={
          Array [
            Object {
              "color": "#000",
              "fontFamily": "System",
              "fontSize": 16,
            },
            Object {
              "fontSize": 14,
              "fontWeight": "bold",
              "paddingLeft": 10,
              "paddingRight": 10,
              "paddingTop": 14,
            },
          ]
        }
        uppercase={false}
      >
        このサンプルの利用について
      </Text>
      <Text
        note={true}
        style={
          Array [
            Object {
              "color": "#a7a7a7",
              "fontFamily": "System",
              "fontSize": 14,
            },
            Object {
              "fontSize": 13,
              "padding": 10,
            },
          ]
        }
        uppercase={false}
      >
        このサンプルを利用する際の注意事項等を記載しています。とはいっても免責事項みたいなものはないのでお気軽にどうぞ！
      </Text>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={null}
    >
      <View
        noBorder={true}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "borderBottomWidth": null,
            "borderColor": "#c9c9c9",
            "flexDirection": "row",
            "marginLeft": 16,
            "paddingBottom": 13,
            "paddingRight": 16,
            "paddingTop": 13,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": null,
              "alignSelf": null,
              "flex": 1,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#000",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "marginLeft": 10,
                  "marginRight": 10,
                },
                Object {
                  "color": "#555555",
                  "fontSize": 14,
                  "fontWeight": "bold",
                  "paddingTop": 2,
                },
              ]
            }
            uppercase={false}
          >
            Q. このサンプルに改変や改良をしても大丈夫ですか？
          </Text>
          <Text
            note={true}
            style={
              Array [
                Object {
                  "color": "#808080",
                  "fontFamily": "System",
                  "fontSize": 14,
                  "fontWeight": "200",
                  "marginLeft": 10,
                  "marginRight": 10,
                },
                Object {
                  "fontSize": 13,
                  "paddingBottom": 8,
                  "paddingTop": 8,
                },
              ]
            }
            uppercase={false}
          >
            A. ご自分でデザインやアカウントが必要な部分を改変した上でご活用頂けましたら、特に問題はないかと思います。もし自分の環境で動かなかった等があればGithubのissue等へ記載をしていただければ幸いです。しかしながら回答によってはご対応が出来かねます場合があることも了承頂ければと思います。
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
